specs/online_order_spec.rb

```diff
@@ -1,90 +1,115 @@
 require 'minitest/autorun'
 require 'minitest/reporters'
 require 'minitest/skip_dsl'
+Minitest::Reporters.use!
 
 # TODO: uncomment the next line once you start wave 3
-# require_relative '../lib/online_order'
+ require_relative '../lib/onlineorder'
 # You may also need to require other classes here
 
 # Because an OnlineOrder is a kind of Order, and we've
 # already tested a bunch of functionality on Order,
 # we effectively get all that testing for free! Here we'll
 # only test things that are different.
 
-xdescribe "OnlineOrder" do
+describe "OnlineOrder" do
   describe "#initialize" do
     it "Is a kind of Order" do
       # Check that an OnlineOrder is in fact a kind of Order
 
       # Instatiate your OnlineOrder here
       # online_order =
       # online_order.must_be_kind_of Grocery::Order
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org", {:street=>"71596 Eden Route",
+        :city=>"Connellymouth", :state=>"LA", :zip=>"98872-9105"})
+
+      online_order = Grocery::OnlineOrder.new(1, products, customer, :complete)
+
+
     end
 
     it "Can access Customer object" do
       # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org",
+        {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip=>"98872-9105"})
+
+      online_order = Grocery::OnlineOrder.new(1, products, customer, :complete)
+
+      online_order.customer.must_be_instance_of Grocery::Customer
     end
 
     it "Can access the online order status" do
       # TODO: Your test code here!
-    end
-  end
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org",
+        {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip=>"98872-9105"})
 
-  describe "#total" do
-    it "Adds a shipping fee" do
-      # TODO: Your test code here!
-    end
+      online_order = Grocery::OnlineOrder.new(1, products, customer, :complete)
 
-    it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      online_order.status.must_equal :complete
     end
   end
 
-  describe "#add_product" do
-    it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
-    end
 
-    it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
-    end
-  end
 
   describe "OnlineOrder.all" do
     it "Returns an array of all online orders" do
       # TODO: Your test code here!
+      result = Grocery::OnlineOrder.all
+      result.must_be_instance_of Array
```

Comment:
You should also check that all elements of the Array or instances of OnlineOrder